vector/src/main/java/im/vector/app/core/di/MavericksViewModelModule.kt

```diff
@@ -505,8 +505,8 @@ interface MavericksViewModelModule {
 
     @Binds
     @IntoMap
-    @MavericksViewModelKey(TextComposerViewModel::class)
-    fun textComposerViewModelFactory(factory: TextComposerViewModel.Factory): MavericksAssistedViewModelFactory<*, *>
+    @MavericksViewModelKey(RoomDetailViewModel::class)
+    fun roomDetailViewModelFactory(factory: RoomDetailViewModel.Factory): MavericksAssistedViewModelFactory<*, *>
```

Comment:
This is not replaced by `MessageComposerViewModel` ?

Reply:
unfortunately not :( when a view model is not in this module it's because it needs a workaround https://github.com/vector-im/element-android/blob/develop/vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt#L161

Usually we can do...

```
companion object : MavericksViewModelFactory<*ViewModel, *ViewState> by hiltMavericksViewModelFactory()
```

which is where the `MavericksViewModelModule` comes in and injects for us

_buttttt_  the voice helpers are marked as `ActivityScoped` which hilt doesn't let us inject into the `ViewModels` as they're some form of singleton scoped~, now that the voice helpers have moved to the `MessageComposerViewModel` we can add the `RoomDetailViewModel` to the normal way of providing the `ViewModels` 

The workaround involves injecting the assisted factory into the fragment and then using that field to provide the `ViewModel`

```
@JvmStatic
override fun create(viewModelContext: ViewModelContext, state: *ViewState): *ViewModel {
  val fragment: *Fragment = (viewModelContext as FragmentViewModelContext).fragment()
  return fragment.factory.create(state)
}
```